File: kotlin/ds/stack.kt
class Stack<T>{
    val items: MutableList<T> = mutableListOf()
    fun isEmpty() = items.isEmpty()
    fun count() = items.size
    fun push(item: T) = items.add(item)
    fun pop() : T? {
        val item = items.lastOrNull()
        if (!isEmpty()){
            items.removeAt(items.size -1)
        }
        return item
    }
    fun peek() : T? = items.lastOrNull()

    override fun toString(): String = items.toString()
}

File: go/ds/stack.go
package main

import "fmt"

type Stack []int

func main() {
	var st Stack
	for i := 0; i <= 10; i++ {
		st.push(i)
	}
	fmt.Println(st)
	for i := 0; i <= 10; i++ {
		st.pop()
	}
	fmt.Println(st)
}

func (s *Stack) push(x int) {
	*s = append(*s, x)
}

func (s *Stack) pop() int {
	if s.isEmpty() {
		return 0
	}
	d := (*s)[len(*s)-1]
	(*s) = (*s)[:len(*s)-1]
	return d
}

func (s *Stack) isEmpty() bool {
	return len(*s) == 0
}
